src/words/environment/WordsClass.java

```diff
@@ -6,12 +7,18 @@
 	private HashMap<String, WordsCustomActionDefinition> functions;
 	private WordsClass parent;
 	private String className;
+	private ArrayList<WordsClass> children;
 	
 	public WordsClass(String className, WordsClass parent) {
 		properties = new HashMap<String, WordsProperty>();
 		functions = new HashMap<String, WordsCustomActionDefinition>();
+		children = new ArrayList<WordsClass>();
 		this.parent = parent;
 		this.className = className;
+		if (parent != null) {
+			// only "thing" should get here
```

Comment:
I'm confused by this comment.  If parent != null, then shouldn't everything *except* "thing" get here?